thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/ThirdEyeDashboardApplication.java

```diff
@@ -40,20 +39,16 @@ public void run(ThirdEyeDashboardConfiguration config, Environment env)
       throws Exception {
     super.initDAOs();
     try {
-      ThirdEyeCacheRegistry.initializeCaches(config, webappConfigDAO);
+      ThirdEyeCacheRegistry.initializeCaches(config);
     } catch (Exception e) {
       LOG.error("Exception while loading caches", e);
     }
     env.jersey().register(new AnomalyFunctionResource(config.getFunctionConfigPath()));
-    env.jersey().register(new DashboardResource(webappConfigDAO));
+    env.jersey().register(new DashboardResource());
```

Comment:
one suggestion is to create daoregistry here and pass it to all resources. Helps in writing test cases